### ISSUE TITLE
install(soure==path) fails if module was already "installed" before

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#383]
I have in annex.py  `initiate_handle` following use of install
```python
                # Also "register" as a sub-handle
                out = install(
                        dataset=Dataset(self.repo.path),
                        path=data_['handle_path'],
                        source=data_['handle_path'],
                        )
```
and it was previously initiated already but then manually 'rm'ed -- so it is present in  the `.gitmodules`.  While running crawler it then attempts to "install" it again but it leads to install puking (see below).  IMHO should be fixed at the level of install.  I.e. if submodule already was registered and points to the same location -- all kosher, just proceed.  If was registered but points to another location (`source` differs) -- might as well puke.  Alternatively (and how I guess I will do now as a workaround) I can skip "install" if already known to the repo submodules
```
(git)smaug:/mnt/btrfs/datasets/datalad/crawl/openfmri[master]git
$> datalad --dbg -l info crawl
2016-03-24 16:31:07,970 [INFO   ] Loading pipeline specification from ./.datalad/crawl/crawl.cfg (crawl.py:89)
2016-03-24 16:31:11,695 [INFO   ] Creating a pipeline with kwargs {} (openfmri_collection.py:33)
2016-03-24 16:31:11,696 [INFO   ] Running pipeline [<datalad.crawler.nodes.crawl_url.crawl_url object at 0x7fbed8282390>, <datalad.crawler.nodes.matches.a_href_match object at 0x7fbed8282410>, assign(assignments={'handle_name': '%(dataset)s'}, interpolate=True), <function _initiate_handle at 0x7fbed82607d0>] (crawl.py:92)
2016-03-24 16:31:11,697 [INFO   ] Fetching 'https://openfmri.org/dataset/' (base.py:429)
2016-03-24 16:31:13,464 [INFO   ] Skipping handle ds000011 since already exists (annex.py:196)
2016-03-24 16:31:13,867 [INFO   ] Skipping handle ds000051 since already exists (annex.py:196)
2016-03-24 16:31:14,138 [INFO   ] Skipping handle ds000052 since already exists (annex.py:196)
2016-03-24 16:31:14,364 [INFO   ] Skipping handle ds000005 since already exists (annex.py:196)
2016-03-24 16:31:14,574 [INFO   ] Skipping handle ds000003 since already exists (annex.py:196)
2016-03-24 16:31:14,821 [INFO   ] Skipping handle ds000009 since already exists (annex.py:196)
2016-03-24 16:31:15,071 [INFO   ] Skipping handle ds000006 since already exists (annex.py:196)
2016-03-24 16:31:15,387 [INFO   ] Initiating handle ds000017 (annex.py:115)
2016-03-24 16:31:16,747 [INFO   ] Skipping handle ds000001 since already exists (annex.py:196)
2016-03-24 16:31:17,119 [INFO   ] Skipping handle ds000007 since already exists (annex.py:196)
2016-03-24 16:31:17,513 [INFO   ] Skipping handle ds000101 since already exists (annex.py:196)
2016-03-24 16:31:17,882 [INFO   ] Skipping handle ds000102 since already exists (annex.py:196)
2016-03-24 16:31:18,251 [INFO   ] Skipping handle ds000105 since already exists (annex.py:196)
2016-03-24 16:31:18,640 [INFO   ] Initiating handle ds000107 (annex.py:115)
2016-03-24 16:31:20,038 [INFO   ] Skipping handle ds000109 since already exists (annex.py:196)
2016-03-24 16:31:20,429 [INFO   ] Skipping handle ds000110 since already exists (annex.py:196)
2016-03-24 16:31:20,770 [INFO   ] Initiating handle ds000113 (annex.py:115)
2016-03-24 16:31:22,351 [ERROR  ] stderr| '' already exists in the index (cmd.py:351)
2016-03-24 16:31:22,351 [ERROR  ] Failed to run ['git', 'submodule', 'add', u'./ds000113', '.'] under u'/mnt/btrfs/datasets/datalad/crawl/openfmri/ds000113'. Exit code=1. out= err='' already exists in the index
|  (cmd.py:295)
Traceback (most recent call last):
  File "/home/yoh/proj/datalad/datalad/venv-tests/bin/datalad", line 9, in <module>
    load_entry_point('datalad==0.1.dev0', 'console_scripts', 'datalad')()
  File "/home/yoh/proj/datalad/datalad/datalad/cmdline/main.py", line 216, in main
    ret = cmdlineargs.func(cmdlineargs)
  File "/home/yoh/proj/datalad/datalad/datalad/interface/base.py", line 153, in call_from_parser
    return cls.__call__(**kwargs)
  File "/home/yoh/proj/datalad/datalad/datalad/interface/crawl.py", line 96, in __call__
    run_pipeline(pipeline)
  File "/home/yoh/proj/datalad/datalad/datalad/crawler/pipeline.py", line 100, in run_pipeline
    output = list(xrun_pipeline(*args, **kwargs))
  File "/home/yoh/proj/datalad/datalad/datalad/crawler/pipeline.py", line 180, in xrun_pipeline
    for idata_out, data_out in enumerate(xrun_pipeline_steps(pipeline, data_in, output=output_sub)):
  File "/home/yoh/proj/datalad/datalad/datalad/crawler/pipeline.py", line 268, in xrun_pipeline_steps
    for data_out in xrun_pipeline_steps(pipeline_tail, data_, output=output):
  File "/home/yoh/proj/datalad/datalad/datalad/crawler/pipeline.py", line 268, in xrun_pipeline_steps
    for data_out in xrun_pipeline_steps(pipeline_tail, data_, output=output):
  File "/home/yoh/proj/datalad/datalad/datalad/crawler/pipeline.py", line 268, in xrun_pipeline_steps
    for data_out in xrun_pipeline_steps(pipeline_tail, data_, output=output):
  File "/home/yoh/proj/datalad/datalad/datalad/crawler/pipeline.py", line 252, in xrun_pipeline_steps
    for data_ in data_in_to_loop:
  File "/home/yoh/proj/datalad/datalad/datalad/crawler/nodes/annex.py", line 1121, in _initiate_handle
    source=data_['handle_path'],
  File "/home/yoh/proj/datalad/datalad/datalad/distribution/install.py", line 215, in __call__
    recursive=recursive)
  File "/home/yoh/proj/datalad/datalad/datalad/distribution/install.py", line 226, in __call__
    runner.run(cmd_list, cwd=ds.path)
  File "/home/yoh/proj/datalad/datalad/datalad/cmd.py", line 296, in run
    raise CommandError(str(cmd), msg, status, out[0], out[1])
CommandError: CommandError: command '['git', 'submodule', 'add', u'./ds000113', '.']' failed with exitcode 1.
Failed to run ['git', 'submodule', 'add', u'./ds000113', '.'] under u'/mnt/btrfs/datasets/datalad/crawl/openfmri/ds000113'. Exit code=1. out= err='' already exists in the index

()
> /home/yoh/proj/datalad/datalad/datalad/cmd.py(296)run()
-> raise CommandError(str(cmd), msg, status, out[0], out[1])
(Pdb) 


$> grep ds000113 .gitmodules 
[submodule "ds000113"]
	path = ds000113
	url = /mnt/datasets/datalad/crawl/openfmri/ds000113
[submodule "ds000113b"]
	path = ds000113b
	url = /mnt/datasets/datalad/crawl/openfmri/ds000113b
```

**Comment by yarikoptic** [comment]
actually situation is more interesting -- apparently .gitmodules  is not consulted really but git modules at this point, so there is no ds000113 in listed by git submodules:
```
$> git submodule | grep 113
 208f2735601168d11680eacc842395b09765ebe2 ds000113b (heads/master)
```
so I can't even workaround quickly based on `data['handle_name'] not in ds.get_dataset_handles()`

**Comment by yarikoptic** [comment]
Addressed in ff8e232
